contracts/Removal.sol

```diff
@@ -80,57 +110,138 @@ contract Removal is
     return removalId.unpackRemovalIdV0();
   }
 
+  // TODO do we want a batch version of this?
+  /**
+   * @notice Get the restriction schedule id (which is the removal's project id) for a given removal id.
+   */
+  function getProjectIdForRemoval(uint256 removalId)
+    public
+    view
+    returns (uint256)
+  {
+    return _removalIdToProjectId[removalId];
+  }
+
+  /**
+   * @notice Get the restriction schedule data for a given removal id.
+   */
+  function getScheduleDataForRemovalId(uint256 removalId)
+    external
+    view
+    returns (ScheduleData memory)
+  {
+    return _projectIdToScheduleData[_removalIdToProjectId[removalId]];
+  }
+
+  /**
+   * @notice Get the restriction schedule data for a given project id.
+   */
+  function getScheduleDataForProjectId(uint256 projectId)
+    external
+    view
+    returns (ScheduleData memory)
+  {
+    return _projectIdToScheduleData[projectId];
+  }
+
   /**
    * @dev mints multiple removals at once (for a single supplier).
    * If `list` is true in the decoded BatchMintRemovalsData, also lists those removals for sale in the market.
    * amounts: [100 * (10 ** 18), 10 * (10 ** 18), 50 * (10 ** 18)] <- 100 tonnes, 10 tonnes, 50 tonnes in standard erc20 units (wei)
    * token id 0 URI points to vintage information (e.g., 2018) nori.com/api/removal/0 -> { amount: 100, supplier: 1, vintage: 2018, ... }
    * token id 1 URI points to vintage information (e.g., 2019) nori.com/api/removal/1 -> { amount: 10, supplier: 1, vintage: 2019, ... }
    * token id 2 URI points to vintage information (e.g., 2020) nori.com/api/removal/2 -> { amount: 50, supplier: 1, vintage: 2020, ... }
+   *
    * @param to The supplier address
    * @param amounts Each removal's tonnes of CO2 formatted as wei
    * @param ids The token ids to use for this batch of removals. The id itself encodes the supplier's ethereum address, a parcel identifier,
-   * the vintage, country code, state code, methodology identifer, and methodology version.
-   * @param data Encodes the market contract address and a unique identifier for the parcel from whence these removals came.
+   * the vintage, country code, state code, methodology identifer, methodology version, and id format.
+   * @param data Encodes the project id and schedule start time for this batch of removals, the market contract
+   * address and a boolean that indicates whether to list these removals for sale now.
    */
   function mintBatch(
     address to,
     uint256[] memory amounts,
     uint256[] memory ids,
     bytes memory data
   ) public override {
+    uint256 idsLength = ids.length;
+    if (!(amounts.length == idsLength)) {
+      revert ArrayLengthMismatch({array1Name: "amounts", array2Name: "ids"});
+    }
+    // todo should we check that the removal id-encoded supplier address for each id is the same as `to` ?
+    // todo should we validate the removal ids any further? what about other malformed fields?
```

Comment:
_**IF**_ we decide to keep doing validation like this as a step in the on-chain mint logic, we shoud definitely extract it to a validate function in `RemovalUtils`

I am starting also starting to think (untested idea) that we could chain the calls for creating an id from on-chain logic in the same tx that we mint so that we don't have to do this validation at all

e.g., [kinda like how this works](https://docs.argent.xyz/wallet-connect-and-argent#multicall)

Reply:
(no action item for this pr-- just following up with context for the TODO)